ast-nodes/parameter.dart

```diff
@@ -9,12 +12,20 @@ class Parameter implements Node {
 
   Parameter(this.name, this.type);
 
-  // TODO: implement .parse()
+  factory Parameter.parse(Iterable<Token> tokens) {
+    var iter = tokens.iterator;
+    checkNext(iter, RegExp('[a-zA-Z_]\w*\$'), "Expected identifier");
```

Comment:
Also, whenever you put backslashes before anything regex-related (\w, \d, \D, etc.), you should double the backslash in strings: `RegExp('[a-zA-Z_]\\w*\$')`. Note that the dollar doesn't need a double backslash because it's a Dart thing, not a regex thing